marconi/src/Marconi/Index/Utxo.hs

```diff
@@ -297,22 +297,24 @@ open dbPath (Depth k) = do
                       ( address TEXT NOT NULL
                       , txId TEXT NOT NULL
                       , txIx INT NOT NULL
-                      , datum TEXT
-                      , datumHash TEXT
+                      , datum BLOB
+                      , datumHash BLOB
                       , value BLOB
-                      , inlineScript TEXT
-                      , inlineScriptHash TEXT
+                      , inlineScript BLOB
+                      , inlineScriptHash BLOB
                       , blockNo INT
                       , slotNo INT
                       , blockHash BLOB
                       , UNIQUE (txId, txIx))|]
   SQL.execute_ c [r|CREATE TABLE IF NOT EXISTS spent
-                      ( txInTxId TEXT PRIMARY KEY NOT NULL UNIQUE
+                      ( txInTxId TEXT PRIMARY KEY NOT NULL
                       , txInTxIx INT NOT NULL
                       , slotNo INT NOT NULL
-                      , blockHash BLOB NOT NULL)|]
+                      , blockHash BLOB NOT NULL
+                      , UNIQUE (txInTxId, txInTxIx))|]
```

Comment:
I remember one argument against this solution was that it would cover up a bug / inefficiency. Having said that, have no idea if there would be any noticeable trade off.

Reply:
This would prevent duplicates in the table, but given that the rows are added with `INSERT OR REPLACE` then it won't raise an exception. I wonder if using just `INSERT` would be better, as you'll get an exception when resuming isn't implemented correctly?

Reply:
It would not raise an exception, and I agree that on an incorrect resume, we wont get an exception. However, in such case, wouldn't the overriding row, be identical to the one what was already there?

Reply:
@kayvank The potential error is that you won't know when you are overwriting old data, and this might hurt when it happens accidentally.